adapters/huaweiads/huaweiads.go

```diff
@@ -0,0 +1,1028 @@
+		}
+
+		adslot30, err := getHuaweiAdsReqAdslot30(huaweiAdsImpExt, &openRTBRequest.Imp[index], openRTBRequest)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+
+		multislot = append(multislot, adslot30)
+	}
+	request.Multislot = multislot
+
+	if err := getHuaweiAdsReqJson(&request, openRTBRequest, huaweiAdsImpExt); err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	//	our request header's Authorization is changing by time, cannot verify by a certain string,
+	//	use isAddAuthorization = false only when run testcase,
+	//	isAddAuthorization = true when we request for a new ad from Huawei Ads ADX.
+	var isAddAuthorization = true
+	if huaweiAdsImpExt.IsAddAuthorization == "false" {
+		isAddAuthorization = false
+	}
+	header = getHeaders(huaweiAdsImpExt, openRTBRequest, isAddAuthorization)
+	bidRequest := &adapters.RequestData{
+		Method:  http.MethodPost,
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: header,
+	}
+
+	var bidRequestArray = make([]*adapters.RequestData, 0, DefaultBidRequestArrayLength)
+	bidRequestArray = append(bidRequestArray, bidRequest)
+
+	return bidRequestArray, errs
+}
+
+func (a *HuaweiAdsAdapter) MakeBids(openRTBRequest *openrtb2.BidRequest, requestToBidder *adapters.RequestData,
+	bidderRawResponse *adapters.ResponseData) (bidderResponse *adapters.BidderResponse, errs []error) {
+	if bidderRawResponse.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	httpStatusError := checkRespStatusCode(bidderRawResponse)
+	if httpStatusError != nil {
+		return nil, []error{httpStatusError}
+	}
+
+	var huaweiAdsResponse HuaweiAdsResponse
+	if err := json.Unmarshal(bidderRawResponse.Body, &huaweiAdsResponse); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "Bad Server Response",
+		}}
+	}
+
+	huaweiAdsResponseError := checkHuaweiAdsResponseRetcode(huaweiAdsResponse)
+	if huaweiAdsResponseError != nil {
+		return nil, []error{huaweiAdsResponseError}
+	}
+
+	var err error
+	bidderResponse, err = convertHuaweiAdsResp2BidderResp(&huaweiAdsResponse, openRTBRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return bidderResponse, nil
+}
+
+// Builder builds a new instance of the HuaweiAds adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &HuaweiAdsAdapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+// get request header
+func getHeaders(huaweiAdsImpExt *openrtb_ext.ExtImpHuaweiAds, request *openrtb2.BidRequest, isAddAuthorization bool) http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	if isAddAuthorization {
+		headers.Add("Authorization", getDigestAuthorization(huaweiAdsImpExt))
+	}
```

Comment:
yes, isAddAuthorization = false only when run testcase, because our request header's Authorization is changing by time, cannot verify by a certain string; It is always true in actual use